device/iot-device-client/src/main/java/com/microsoft/azure/sdk/iot/device/transport/amqps/AmqpsIotHubConnection.java

```diff
@@ -740,35 +740,13 @@ else if (this.connection.getLocalState() == EndpointState.CLOSED && this.connect
         }
     }
 
-    private void executorServicesCleanup() throws TransportException
+    private void executorServicesCleanup()
     {
         if (this.executorService != null)
         {
             log.trace("Shutdown of executor service has started");
-            this.executorService.shutdown();
-            try
-            {
-                // Wait a while for existing tasks to terminate
-                if (!this.executorService.awaitTermination(MAX_WAIT_TO_TERMINATE_EXECUTOR, TimeUnit.SECONDS))
-                {
-                    this.executorService.shutdownNow(); // Cancel currently executing tasks
-                    // Wait a while for tasks to respond to being cancelled
-                    if (!this.executorService.awaitTermination(MAX_WAIT_TO_TERMINATE_EXECUTOR, TimeUnit.SECONDS))
```

Comment:
it looks like we were trying really hard to make sure the executorService gets terminated before nulling it out. 
What are the consequences of not waiting for these anymore?

Reply:
Nothing really. This executor was used to manage the one worker thread that we extend to proton-j to run the AMQP connection. When this function is called, we are deliberately closing everything, including this thread pool.

Reply:
For reference, this method is only called when opening the connection fails or when we have finished closing a successful AMQP connection. In both cases, there is nothing left that we need to be careful about cleaning up in the AMQP library.